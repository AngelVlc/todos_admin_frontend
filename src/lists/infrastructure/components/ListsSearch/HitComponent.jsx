import React from "react";
import { useHistory } from "react-router-dom";
import { Highlight } from 'react-instantsearch';

export const HitComponent = ({hit}) => {
  let history = useHistory();

  return (
    <article onClick={() => history.push(`/lists/${hit.objectID}`)}>
      <Highlight attribute="name" hit={hit} />
    </article>
  )
}