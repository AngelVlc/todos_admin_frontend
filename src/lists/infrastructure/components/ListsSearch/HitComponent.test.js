import { render, cleanup } from "@testing-library/react";
import { HitComponent } from "./HitComponent";

describe("HitComponent", () => {
  const renderComponent = (hit) => {
    return {
      ...render(<HitComponent hit={hit} />),
    };
  };

  afterEach(cleanup);

  it("should match the snapshot", () => {
    const hit = {
      userID: 1,
      name: "listname",
      itemsTitles: ["title"],
      itemsDescriptions: ["description"],
      objectID: "20",
      _highlightResult: {
        userID: {
          value: "1",
          matchLevel: "none",
          matchedWords: [],
        },
        name: {
          value: "<mark>list</mark>name",
          matchLevel: "full",
          fullyHighlighted: false,
          matchedWords: ["list"],
        },
        itemsTitles: [{ value: "title", matchLevel: "none", matchedWords: [] }],
        itemsDescriptions: [
          { value: "description", matchLevel: "none", matchedWords: [] },
        ],
      },
      __position: 1,
    };

    const { asFragment } = renderComponent(hit);

    expect(asFragment()).toMatchSnapshot();
  });
});
