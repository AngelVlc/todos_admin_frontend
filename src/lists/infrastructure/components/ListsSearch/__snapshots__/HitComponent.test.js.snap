// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HitComponent should match the snapshot 1`] = `
<DocumentFragment>
  <article>
    <span
      class="ais-Highlight"
    >
      <mark
        class="ais-Highlight-highlighted"
      >
        list
      </mark>
      <span
        class="ais-Highlight-nonHighlighted"
      >
        name
      </span>
    </span>
  </article>
</DocumentFragment>
`;
